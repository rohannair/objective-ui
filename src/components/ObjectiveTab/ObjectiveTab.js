import React, { PropTypes } from 'react'
import dateformat from 'dateformat'
import Pill from '../Pill'
import { toMonthDayHour } from '../../utils/dates'
import SplitIconPill from '../SplitIconPill'

const ObjectiveTab = p => (
  <div style={{
    display: 'flex',
    alignItems: 'baseline'
  }}
  className='objectiveTab'
  >
  	{ p.objective
      ? (
        	<SplitIconPill readOnly={p.readOnly} info iconAction={p.editObjective} icon='edit'>{p.objective.name}</SplitIconPill>
        )
  	  : (
          <SplitIconPill transparent action={p.editObjective} iconAction={p.editObjective} icon='plus'>Add Objective</SplitIconPill>
        )
    }
	{ p.blocker && <Pill danger><i className={'zmdi zmdi-alert-circle-o'} /> BLOCKER!</Pill>}
	<small>{toMonthDayHour(p.createdAt)}</small>
  </div>
)

export default ObjectiveTab
