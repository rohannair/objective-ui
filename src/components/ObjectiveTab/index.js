import ObjectiveTab from './ObjectiveTab'
export default ObjectiveTab
