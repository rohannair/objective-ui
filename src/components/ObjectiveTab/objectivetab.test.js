import React from 'react'

import ObjectiveTab from './index.js'

describe('Test suite for ObjectiveTab component', () => {
  it('ObjectiveTab should exist', () => {
    let wrapper = shallow(<ObjectiveTab/>)
    expect(wrapper).toMatchSnapshot()
  })
})

