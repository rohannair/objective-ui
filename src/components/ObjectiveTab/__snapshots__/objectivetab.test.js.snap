exports[`Test suite for ObjectiveTab component ObjectiveTab should exist 1`] = `
<ObjectiveTab
  className="iPXcnY" />
`;
